chain/xrp/client/client.go

```diff
@@ -0,0 +1,992 @@
+					newBalance, parseErr = strconv.ParseFloat(node.CreatedNode.NewFields.Balance.XRPAmount, 10)
+					if parseErr != nil {
+						return xclient.TxInfo{}, parseErr
+					}
+
+				} else {
+					decimals = 15
+					newBalance, parseErr = strconv.ParseFloat(node.CreatedNode.NewFields.Balance.TokenAmount.Value, 10)
+					if parseErr != nil {
+						return xclient.TxInfo{}, parseErr
+					}
+
+					if newBalance > 0 {
+						transactionType = "destination"
+						contract = node.CreatedNode.NewFields.HighLimit.Issuer
+					} else {
+						transactionType = "source"
+						contract = node.CreatedNode.NewFields.LowLimit.Issuer
+					}
+				}
+
+				var newBalanceString string
+				if decimals == 0 {
+					newBalanceString = fmt.Sprintf("%.6f", math.Abs(newBalance))
+				} else {
+					newBalanceString = fmt.Sprintf("%.15f", math.Abs(newBalance))
+				}
+
+				newBalanceString = strings.TrimRight(strings.TrimRight(newBalanceString, "0"), ".")
+
+				convertedTransactedAmount, conversionErr := xc.NewAmountHumanReadableFromStr(newBalanceString)
+				if conversionErr != nil {
+					return xclient.TxInfo{}, conversionErr
+				}
+
+				if transactionType == "destination" {
+					address = xc.Address(createdNode.NewFields.LowLimit.Issuer)
+				} else if transactionType == "source" {
+					address = xc.Address(createdNode.NewFields.HighLimit.Issuer)
+				}
+
+				tf.AddDestination(
+					address,
+					xc.ContractAddress(contract),
+					convertedTransactedAmount.ToBlockchain(decimals),
+					nil,
+				)
+			}
+		}
+	}
+
+	txInfo.AddTransfer(tf)
+
+	txInfo.Fees = txInfo.CalculateFees()
+
+	return txInfo, nil
+}
+
+func extractModifiedNodeBalance(node *ModifiedNode) (xc.AmountBlockchain, error) {
+	var (
+		finalBalance, previousBalance float64
+		decimals                      int32
+		parseErr                      error
+	)
+
+	if node.FinalFields.Balance.XRPAmount != "" {
+		decimals = 0
+		finalBalance, parseErr = strconv.ParseFloat(node.FinalFields.Balance.XRPAmount, 10)
+	} else {
+		decimals = 15
+		finalBalance, parseErr = strconv.ParseFloat(node.FinalFields.Balance.TokenAmount.Value, 10)
+	}
+	if parseErr != nil {
+		return xc.AmountBlockchain{}, parseErr
+	}
+
+	if node.PreviousFields.Balance.XRPAmount != "" {
+		previousBalance, parseErr = strconv.ParseFloat(node.PreviousFields.Balance.XRPAmount, 10)
+	} else {
+		previousBalance, parseErr = strconv.ParseFloat(node.PreviousFields.Balance.TokenAmount.Value, 10)
+	}
+	if parseErr != nil {
+		return xc.AmountBlockchain{}, parseErr
+	}
+
+	transactedAmount := math.Abs(previousBalance - finalBalance)
+
+	var transactedAmountString string
+	if decimals == 0 {
+		transactedAmountString = fmt.Sprintf("%.6f", transactedAmount)
+	} else {
+		transactedAmountString = fmt.Sprintf("%.15f", transactedAmount)
+	}
+
+	transactedAmountString = strings.TrimRight(strings.TrimRight(transactedAmountString, "0"), ".")
+
+	convertedTransactedAmount, conversionErr := xc.NewAmountHumanReadableFromStr(transactedAmountString)
```

Comment:
so we are going from `string -> float -> string -> decimal`?  Let's do `string -> decimal`.